scripts/claude_login.sh

```
@@ -0,0 +1,209 @@
+
+        # Remove 'encrypted:' prefix if present
+        api_key="${OPENAI_API_KEY#encrypted:}"
+
+        # Test OpenAI API (if it's an OpenAI key)
+        if [[ "$api_key" == sk-* ]]; then
+            echo "Testing OpenAI API connection..."
+            # Use curl with timeout flags and separate status code capture
+            # Write body to a temporary variable and status code separately
+            temp_body=$(mktemp)
+            # Install trap to ensure temp file is cleaned up on exit or signals
+            trap "rm -f '$temp_body'" EXIT INT TERM
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)